pkg-manager/resolve-dependencies/test/validatePeerDependencies.test.ts

```diff
@@ -0,0 +1,27 @@
+import { validatePeerDependencies } from '../src/validatePeerDependencies'
+
+test('accepts valid specifications that make sense for peerDependencies', () => {
+  validatePeerDependencies({
+    'use-version-range': '>=1.2.3 || ^3.2.1',
+    'use-workspace': 'workspace:^',
+    'use-catalog': 'catalog:',
```

Comment:
These are only valid for packages from the workspace but you validate all dependencies at the moment. Maybe just validate packages from the workspace.

Reply:
I have moved the call to `validatePeerDependencies` to `toResolveImporter`: https://github.com/pnpm/pnpm/pull/8942/commits/924a8683f24cbb272f21c90eb1d4cb935de2be7c